VideoLocker/src/androidTest/java/org/edx/mobile/test/feature/interactor/RegistrationScreenInteractor.java

```diff
@@ -0,0 +1,101 @@
+
+    public RegistrationScreenInteractor observeRegistrationScreen() {
+        final MainApplication app = MainApplication.instance();
+        try {
+            registrationFieldsDesc = app.getInjector().getInstance(EdxEnvironment.class).getServiceManager().getRegistrationDescription();
+        } catch (Exception e) {
+            throw new AssertionError("Failed to load registration form fields json");
+        }
+        final CharSequence title = ResourceUtil.getFormattedString(app.getResources(), R.string.register_title, "platform_name", app.getInjector().getInstance(Config.class).getPlatformName());
+        onView(withId(R.id.activity_title)).check(matches(withText(title.toString())));
+        onEmailView().check(matches(isCompletelyDisplayed()));
+        onNameView().check(matches(isCompletelyDisplayed()));
+        onUsernameView().check(matches(isCompletelyDisplayed()));
+        onPasswordView().check(matches(isCompletelyDisplayed()));
+        onCountryView().check(matches(isCompletelyDisplayed()));
+        onCreateAccountButton().check(matches(isCompletelyDisplayed()));
+        return this;
+    }
+
+    public MyCoursesScreenInteractor createAccount(Credentials credentials) {
+        onEmailView().perform(typeText(credentials.email), closeSoftKeyboard());
+        onNameView().perform(typeText("Test Account"), closeSoftKeyboard());
+        onUsernameView().perform(typeText(credentials.username), closeSoftKeyboard());
+        onPasswordView().perform(typeText(credentials.password), closeSoftKeyboard());
+        onCountryView().perform(SpinnerViewAction.selectSpinnerItem(1));
+        onCreateAccountButton().perform(click());
+        return new MyCoursesScreenInteractor();
+    }
+
+    private ViewInteraction onEmailView() {
+        return onView(withHint(findHintByName("email")));
+    }
+
+    private ViewInteraction onNameView() {
+        return onView(withHint(findHintByName("name")));
+    }
+
+    private ViewInteraction onUsernameView() {
+        return onView(withHint(findHintByName("username")));
+    }
+
+    private ViewInteraction onPasswordView() {
+        return onView(withHint(findHintByName("password")));
+    }
+
+    private ViewInteraction onCountryView() {
+        return onView(withText(findHintByName("country")));
+    }
+
+    private ViewInteraction onCreateAccountButton() {
+        return onView(withId(R.id.create_account_tv));
+    }
+
+    /**
+     * Registration fields are generated at runtime using the
+     * `config/registration_form.json` file. The `label` field
+     * in the json file is translated as the hint for input
+     * views on registration screen.
+     *
+     * @param name Value of the `name` field in json object.
+     * @return Hint of the input view.
+     */
+    @NonNull
+    private String findHintByName(@NonNull String name) {
+        for (RegistrationFormField formField : registrationFieldsDesc.getFields()) {
```

Comment:
Actually, thinking a bit more on this, we're defining variable input fields for the actual sign-up screen, but assuming the existence of some specific fields for the test here. It works for our specific configuration, but if someone excludes e.g. the email or username field, then this test would fail. The right thing to do seems to be to have the test inputs defined as variables as well, with variable key names/types.